commons/src/test/java/io/aiven/kafka/tieredstorage/commons/SegmentCompressionCheckerTest.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tieredstorage.commons;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.TimestampType;
+
+import org.assertj.core.util.Files;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SegmentCompressionCheckerTest {
+
+    @TempDir
+    Path dir;
+
+    @Test
+    void shouldRequireCompression_whenCompressionNone() throws IOException {
+        final File file = Files.newFile(dir.resolve("segment.log").toString());
+        try (final FileRecords records = FileRecords.open(file, false, 100000, true)) {
+            try (final MemoryRecordsBuilder builder = MemoryRecords.builder(
```

Comment:
With try-with-resource I don't see the issue having nested `try`s.
Are you suggesting to put all resources on the same try? It may save us a line or two but not sure is more readable.

Reply:
I gave a try, lmk if looks good.